File: week5/0169. Majority Element/majority-mlement.py
# 摩尔投票法
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        if not nums:
            return

        if len(nums) == 1:
            return nums[0]
        votes = 0
        for i in nums:
            if votes == 0:
                x =i
            if i == x:
                votes +=1
            else:
                votes -=1
        return x

File: week5/0169. Majority Element/给定一个大小为 n 的数组，找到其中的多数元素。多数元素是指在数组中出现次数 大于 ⌊ n/2 ⌋ 的元素。  你可以假设数组是非空的，并且给定的数组总是存在多数元素。     示例 1：  输入：[3,2,3] 输出：3 示例 2：  输入：[2,2,1,1,1,2,2] 输出：2  来源：力扣（LeetCode） 链接：https:/leetcode-cn.com/problems/majority-mlement.go
func majorityElement(nums []int) int {
    votes := 0
    x := 0
    for _,num := range nums{
        if votes == 0{
            x = num
        }
        if num == x{
            votes +=1
        }else{
            votes -=1
        }
    }
    return x

}